File: backend/src/main/java/com/shop/features/product/service/ProductService.java
package com.shop.features.product.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.shop.features.product.dto.ProductFilterDto;
import com.shop.features.product.dto.ProductRequestDto;
import com.shop.features.product.dto.ProductResponseDto;


public interface ProductService {

    ProductResponseDto createProduct(ProductRequestDto request);

    Page<ProductResponseDto> getAllProducts(int page, int size);

    Page<ProductResponseDto> getFilteredProducts(
            ProductFilterDto filter,
            int page,
            int size,
            String sortBy,
            String direction
    );
    ProductResponseDto updateProduct(
            Long productId,
            ProductRequestDto request
    );
    void deleteProduct(Long productId);
    ProductResponseDto getProductById(Long id);
    
    List<ProductResponseDto> getFeaturedProducts();

    List<ProductResponseDto> getNewArrivalProducts();

    List<ProductResponseDto> getBestSellerProducts();

    List<ProductResponseDto> getTrendingProducts();
    
    List<ProductResponseDto> getMyProducts();
}

File: frontend/src/services/productService.js
import api from "./api";

// =============================
// PRODUCT CRUD
// =============================

export const getProducts = async (
  page = 0,
  size = 20
) => {
  const response = await api.get(
    `/products?page=${page}&size=${size}`
  );

  return response.data.data.content;
};

export const getProductById = async (id) => {
  const response = await api.get(
    `/products/${id}`
  );

  return response.data.data;
};

export const createProduct = async (
  productData
) => {
  const response = await api.post(
    "/products",
    productData
  );

  return response.data;
};

export const updateProduct = async (
  id,
  productData
) => {
  const response = await api.put(
    `/products/${id}`,
    productData
  );

  return response.data;
};

export const deleteProduct = async (
  id
) => {
  const response = await api.delete(
    `/products/${id}`
  );

  return response.data;
};

// =============================
// HOME PAGE SECTIONS
// =============================

export const getFeaturedProducts =
  async () => {
    const res = await api.get(
      "/products/featured"
    );

    return res.data;
  };

export const getNewArrivalProducts =
  async () => {
    const res = await api.get(
      "/products/new-arrivals"
    );

    return res.data;
  };

export const getBestSellerProducts =
  async () => {
    const res = await api.get(
      "/products/best-sellers"
    );

    return res.data;
  };

export const getTrendingProducts =
  async () => {
    const res = await api.get(
      "/products/trending"
    );

    return res.data;
  };

  export const getMyProducts = async () => {
  const response = await api.get(
    "/products/my-products"
  );

  return response.data;
};